Terminal.Gui/ConsoleDrivers/NetDriver/NetMainLoop.cs

```diff
@@ -47,55 +46,37 @@ void IMainLoopDriver.Setup (MainLoop mainLoop)
         Task.Run (NetInputHandler, _inputHandlerTokenSource.Token);
     }
 
-    void IMainLoopDriver.Wakeup () { _eventReady.Set (); }
+    void IMainLoopDriver.Wakeup () { }
 
     bool IMainLoopDriver.EventsPending ()
     {
         _waitForProbe.Set ();
 
-        if (_mainLoop.CheckTimersAndIdleHandlers (out int waitTimeout))
+        if (_mainLoop.CheckTimersAndIdleHandlers (out int _))
         {
             return true;
         }
 
-        try
-        {
-            if (!_eventReadyTokenSource.IsCancellationRequested)
-            {
-                // Note: ManualResetEventSlim.Wait will wait indefinitely if the timeout is -1. The timeout is -1 when there
-                // are no timers, but there IS an idle handler waiting.
-                _eventReady.Wait (waitTimeout, _eventReadyTokenSource.Token);
```

Comment:
It may not have input but have idle handler's from synchronization context or Application.Invoke. We must avoid unnecessary iterations to not cause screen flickering.